### ISSUE TITLE
Kyrix spec as JSON schema?

### ISSUE THREAD

**Issue by ricardoekm** [ricardoekm/Kyrix#7]
There's a lot of validation code in Kyrix compiler to check if the right attributes are in the right place with the correct values. E.g.:
https://github.com/ricardoekm/Kyrix/blob/0ce6879a307f1d16440ef79fa978d9dc529020a2/compiler/src/View.js#L10-L19

We could eliminate most of this work if we declare Kyrix spec as a [JSON Schema](https://json-schema.org/understanding-json-schema/)

This also provides the benefit of a live documentation of Kyrix spec.

**Comment by tracyhenry** [comment]
definitely. also tracked here: https://github.com/tracyhenry/Kyrix/issues/105
